File: backend/src/main/java/org/educama/shipment/api/resource/ShipmentResource.java
package org.educama.shipment.api.resource;

import org.educama.enums.ClientType;
import org.educama.shipment.api.datastructure.CargoDS;
import org.educama.shipment.api.datastructure.PartyDS;
import org.educama.shipment.api.datastructure.ServicesDS;
import org.educama.shipment.model.Shipment;

/**
 * REST-Resource for single shipment.
 */
public class ShipmentResource {

    public String trackingId;
    public PartyDS sender;
    public PartyDS receiver;
    public CargoDS shipmentCargo;
    public ServicesDS shipmentServices;
    public ClientType customerTypeEnum;

    /**
     * Create a API-Model (Resource) instance from the internal data model.
     *
     * @param shipmentModel instance of the internal-data model
     * @return a converted ShipmentResource
     */
    public ShipmentResource fromShipment(Shipment shipmentModel) {
        this.trackingId = shipmentModel.trackingId;
        this.sender = new PartyDS(shipmentModel.sender);
        this.receiver = new PartyDS(shipmentModel.receiver);
        this.shipmentCargo = new CargoDS(shipmentModel.shipmentCargo);
        this.shipmentServices = new ServicesDS(shipmentModel.shipmentServices);
        this.customerTypeEnum = shipmentModel.customerTypeEnum;

        return this;
    }
}

File: frontend/src/app/shipment/shipment-common/api/resources/shipment.resource.ts
import {Cargo} from "../datastructures/cargo.datastructure";
import {ShipmentServices} from "../datastructures/services.datastructure";
import {Party} from "../datastructures/party.datastructure";
export class ShipmentResource {
    public trackingId: string;
    public uuidSender: string;
    public uuidReceiver: string;
    public sender: Party;
    public receiver: Party;
    public customerTypeEnum: string;
    public shipmentCargo: Cargo;
    public shipmentServices: ShipmentServices;
}
